src/preview.html

```diff
@@ -1,142 +1,201 @@
 <head>
     <style>
-        html, body, #canvas { margin: 0; padding: 0; width: 100%; height: 100%; display: block; }
-        #message {font-family: Consolas; font-size: 1.2em; color:#ccc; background-color:black; font-weight: bold; z-index: 2; position: absolute;}
+        html, body, #canvas {
+            margin: 0;
+            padding: 0;
+            width: 100%;
+            height: 100%;
+            display: block;
+        }
+        .error {
+            font-family: Consolas;
+            font-size: 1.2em;
+            color: black;
+            box-sizing: border-box;
+            background-color: lightcoral;
+            border-radius: 2px;
+            border-color: lightblue;
+            border-width: thin;
+            border-style: solid;
+            line-height: 1.4em;
+        }
+        .error:hover {
+            color: black;
+            background-color: brown;
+            border-color: blue;
+        }
+        #message {
+            font-family: Consolas;
+            font-size: 1.2em;
+            color: #ccc;
+            background-color: black;
+            font-weight: bold;
+            z-index: 2;
+            position: absolute;
+        }
     </style>
 </head>
 <body>
     <div id="message"></div>
     <div id="container"></div>
 
-    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
-    <script src="https://cdnjs.cloudflare.com/ajax/libs/three.js/84/three.min.js"></script>
-    <canvas id="canvas"></canvas>
-    <script id="vs" type="x-shader/x-vertex">
-        void main() {
-            gl_Position = vec4(position, 1.0);
-        }
-    </script>
-    <script id="fs" type="x-shader/x-fragment">
-        uniform vec3        iResolution;
-        uniform float       iGlobalTime;
-        uniform float       iTimeDelta;
-        uniform int         iFrame;
-        uniform float       iChannelTime[4];
-        uniform vec3        iChannelResolution[4];
-        uniform vec4        iMouse;
-        uniform sampler2D   iChannel0;
-        uniform sampler2D   iChannel1;
-        uniform sampler2D   iChannel2;
-        uniform sampler2D   iChannel3;
-        uniform sampler2D   iChannel4;
-        uniform sampler2D   iChannel5;
-        uniform sampler2D   iChannel6;
-        uniform sampler2D   iChannel7;
-        uniform sampler2D   iChannel8;
-        uniform sampler2D   iChannel9;      
+</body>
+<script src="file://d:\zollk\Documents\Visual Studio 2017\Projects\shader-toy\resources\jquery.min.js"></script>
+<script src="file://d:\zollk\Documents\Visual Studio 2017\Projects\shader-toy\resources\three.min.js"></script>

```

Comment:
Absolute paths aren't going to work here.

Reply:
Yeah, true, my bad. I'm gonna have to fix that when I'm home. Not sure what the original said there, probably had a url to the min.js versions of those.

Reply:
Instead I just copied the result of the whole parsing process.

Reply:
Fixed this and also added local stats.js because I was working mainly without internet and couldn't check performance  :D